### ISSUE TITLE
Maven: transitive dependencies should be considered by AddDependency

### ISSUE THREAD

**Issue by murdos** [openrewrite/rewrite#646]
For example, if I use `spring-boot-starter-test` dependency in my project, I'll get assertj, junit, dependencies.

But if I use `org.openrewrite.java.testing.junit5.JUnit5BestPractices` recipe, that use the `AddDependency` recipe, JUnit dependency will be added to my pom.xml whereas it's already available as a transitive dependency.

**Comment by murdos** [comment]
I can confirm that the original issue is fixed for transitive dependencies declared directly in the module, but it doesn't work if the dependency is declared by the parent module.

Simplified example:
#### Parent module
```xml
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns="http://maven.apache.org/POM/4.0.0"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <artifactId>parent-module</artifactId>
  <packaging>pom</packaging>

  <modules>
    <module>child-module</module>
  </modules>

  <dependencies>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-test</artifactId>
      <scope>test</scope>
    </dependency>
  </dependencies>
</project>
```

#### Child module
```xml
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns="http://maven.apache.org/POM/4.0.0"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <parent>
    <artifactId>parent-module</artifactId>
  </parent>

  <artifactId>child-module</artifactId>
  <packaging>jar</packaging>

</project>
```

JUnit dependency will be added to child module pom.xml whereas it's already available through parent transitive dependency.

**Comment by jkschneider** [comment]
Fixed by #813
